packages/nodes-from-anchor/src/v01/typeNodes/StructFieldTypeNode.ts

```diff
@@ -0,0 +1,26 @@
+import { StructFieldTypeNode, structFieldTypeNode } from '@kinobi-so/nodes';
+
+import { IdlV01Field, IdlV01Type } from '../idl';
+import { typeNodeFromAnchorV01 } from './TypeNode';
+
+export function structFieldTypeNodeFromAnchorV01(idl: IdlV01Field | IdlV01Type): StructFieldTypeNode {
+    const field = idl as IdlV01Field;
+
+    const isFieldDefinition = 'docs' in field && 'name' in field && 'type' in field;
+
+    if (isFieldDefinition) {
+        return structFieldTypeNode({
+            docs: field.docs ?? [],
+            name: field.name,
+            type: typeNodeFromAnchorV01(field.type),
+        });
+    }
+
+    const type = idl as IdlV01Type;
+
+    return structFieldTypeNode({
+        docs: [],
+        name: '',
+        type: typeNodeFromAnchorV01(type),
+    });
```

Comment:
Is this a thing in the new IDL spec? Can field not be named? Or should this be converted into a `TupleTypeNode` instead?

Reply:
```typescript
export type IdlV01TypeDefTyStruct = {
    fields?: IdlV01DefinedFields;
    kind: 'struct';
};

export type IdlV01DefinedFields = IdlV01DefinedFieldsNamed | IdlV01DefinedFieldsTuple;

export type IdlV01DefinedFieldsNamed = IdlV01Field[];

export type IdlV01DefinedFieldsTuple = IdlV01Type[];
```

A single field can now be of type field which aligns with the structFieldTypeNode but the IdlV01Type are simple types like 'string', 'u8', etc

Reply:
What do you think on this? I threw an error but the return type should be StructFieldNode or TypeNode.